data-mongodb/src/main/java/io/micronaut/data/mongodb/repository/MongoQueryExecutor.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.data.mongodb.repository;
+
+import com.mongodb.client.model.DeleteOptions;
+import com.mongodb.client.model.UpdateOptions;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.data.mongodb.operations.options.MongoAggregationOptions;
+import io.micronaut.data.mongodb.operations.options.MongoFindOptions;
+import org.bson.conversions.Bson;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * MongoDB specific repository that allows to use direct BSON objects.
+ *
+ * @param <E> The entity type
+ * @author Denis Stepanov
+ * @since 3.3
+ */
+public interface MongoQueryExecutor<E> {
+
+    /**
+     * Finds one result.
+     *
+     * @param filter The filter to be applied
+     * @return The optional result
+     */
+    Optional<E> findOne(@Nullable Bson filter);
+
+    /**
+     * Finds one result.
+     *
+     * @param options The options
+     * @return The optional result
+     */
+    Optional<E> findOne(MongoFindOptions options);
+
+    /**
+     * Finds all results.
+     *
+     * @param filter The filter to be applied
+     * @return The records
+     */
+    List<E> findAll(@Nullable Bson filter);
+
+    /**
+     * Finds all results.
+     *
+     * @param options The options
+     * @return The records
+     */
+    List<E> findAll(MongoFindOptions options);
```

Comment:
Is the argument `@Nullable`? Add `@NonNull` to return type. 
Wouldn't it be better to return a FindIterable? What about pagination?